### ISSUE TITLE
Migrate from deprecated "openjdk" Docker image

### ISSUE THREAD

**Issue by ryanfb** [DSpace/DSpace#9277]
DSpace currently uses the [`openjdk`](https://hub.docker.com/_/openjdk/) Docker image, which is now officially deprecated:

> This image is officially deprecated and all users are recommended to find and use suitable replacements ASAP. Some examples of other Official Image alternatives (listed in alphabetical order with no intentional or implied preference):
> 
> * [amazoncorretto](https://hub.docker.com/_/amazoncorretto)
> * [eclipse-temurin](https://hub.docker.com/_/eclipse-temurin)
> * [ibm-semeru-runtimes](https://hub.docker.com/_/ibm-semeru-runtimes)
> * [ibmjava](https://hub.docker.com/_/ibmjava)
> * [sapmachine](https://hub.docker.com/_/sapmachine)

While I don't know which image would be the most appropriate choice to migrate to for the whole community, it's clear that a different image needs to be used.

**Comment by tdonohue** [comment]
Based on https://github.com/docker-library/openjdk/issues/505 it sounds like most people are switching the [eclipse-temurin](https://hub.docker.com/_/eclipse-temurin). I have no strong preference but this needs a volunteer.

**Comment by hardyoyo** [comment]
alas, the https://hub.docker.com/_/eclipse-temurin images don't have a "slim" variant listed in their tags... same story for the other listed alternatives.... can we live without a "slim" image?

**Comment by ryanfb** [comment]
For what it's worth, most of the use of the `openjdk` image is as an intermediate build stage used to run `ant` as a build/deploy stage which then has its build artifacts copied into a later build stage - with the exception of `Dockerfile.cli` which uses `openjdk` as the final stage. That is, if part of the reason for using "slim" variants is final image size, differences in size between `openjdk` and what's selected should only really affect the CLI image's size.

**Comment by tdonohue** [comment]
> can we live without a "slim" image?

Yes, we can just chose a different image that looks reasonable for now.  If a better option comes up later, it's not difficult to switch again in the future.

**Comment by misilot** [comment]
For what it's worth the version of openjdk is already Eclipse Adoptium / Temurin-11.0.22+7

<img width="1281" alt="image" src="https://github.com/DSpace/DSpace/assets/1446856/5402784d-ee96-424d-b01c-2f8629473b5a">

I can make a PR with the changes to switch to eclipse-temurin
